src/MCP23017.h

```diff
@@ -11,6 +11,26 @@ enum class MCP23017Port : uint8_t
 	B = 1
 };
 
+enum class MCP23017IOPin : uint8_t
+{

```

Comment:
Given that the existing `MCP23017Port` enum, I'd rather remove the `IO` from this one for consistency 🙂 

```suggestion
enum class MCP23017Pin : uint8_t
{
```

Which leads me to the next question. How do you intend to use it ?
Neither of these examples would build obviously, so despite the fact that I agree with you on the readability/comprehension side, I cannot see how you would use it 🤔 

```cpp
uint8_t pin = MCP23017IOPin::GPB3;
uint8_t value = mcp.digitalRead(pin);
```

```cpp
uint8_t value = mcp.digitalRead(MCP23017IOPin::GPB3);
```

We would need to define custom operators and/or casts, and use `MCP23017Pin` in all methods signatures (which would be a breaking change).

Reply:
I forgot about the specifics of enum class. I checked current discussions of the subject, and implemented a suggested pattern to solve this pre-C++23. What do you think?